js-exercises/telephone-check/telephoneCheck.js

```diff
@@ -0,0 +1,13 @@
+function telephoneCheck(phoneNumber) {
+  if (typeof phoneNumber !== 'string') {
+    throw Error(`Expected String got ${typeof phoneNumber}`);
+  }
+  if (phoneNumber.length < 11) {
+    return false;
+  }
+
+  const pattern = new RegExp(/^(1\s?)?((\([0-9]{3}\))|[0-9]{3})[\s-]?[\0-9]{3}[\s-]?[0-9]{4}$/, 'i');
```

Comment:
can you give an example? @varora1406

Reply:
you can check :
```
const numberPattern = (length) => `[0-9]{${length}}`;
const pattern = new RegExp(`^[\0-9]{3}[\s-]?${numberPattern(4)}$`, 'i');
pattern1.test('555-5555') => results true
```

I couldn't make complete pattern work, otherwise I would have provided you a full suggestion 🙂